### DISCUSSION TITLE
Features to add

### DISCUSSION THREAD

**Discussion by adnanmunawar** [LCSR-SICKKIDS/volumetric_drilling#20] [Ideas]
These are the features we should add:

**Comment by jabarragann** [comment]
Features to improve the volumetric drilling's data recorder

1. Create ambf plugin to record videos using FFmpeg. Videos encoded with FFmpeg will be smaller than those stored in HDF5.
2. All low-dimensional data from the simulator should be kept in the hdf5 file, e.g., force and voxels removed.
3. Restructure headers inside the HDF5 to be vision and kinematics data.
4. Add version information to the HDF5 files, so that all old files can still be accessed/parsed using the old code.
5. Add @nnagururu json metadata the HDF5 metadata. (User ID, anatomy used for experiment)

**Comment by nnagururu** [comment]
Re pupil calibration:
- Need to implement new grid calibration routine
- Automatically generate world video
